Documentation/concepts/kubernetes/configuration.rst

```diff
@@ -47,6 +47,18 @@ to your preferences:
   packets that involve an L4 connection state change. Valid options are
   ``none``, ``low``, ``medium``, ``maximum``.
 
+  - ``none`` - Generate a tracing event on every receive and send packet.
+  - ``low`` - Generate a tracing event on every send packet.
+  - ``medium`` - Generate a tracing event on every new connection and connection
+    state change, and periodically generate tracing events on new sets of TCP
+    flags seen for existing TCP connections. The reporting interval for new TCP
+    flags is defined by ``monitor-aggregation-interval``.
```

Comment:
The new docs are definitely an improvement!  There are still some points of confusion for me though:

1. Imagine we have an established connection.  Then a packet arrives with the PSH flag set (as well as ACK, as normal).  Will this generate a monitor event immediately?  Or will it start a timer and generate an event only after `monitor-aggregation-interval` (covering any/all PSH packets received during that interval)?
2. Now imagine that `monitor-aggregation-interval` elapses, and another PSH,ACK packet arrives.  Does this generate a new notification?  In other words, does "new flags" mean "new over the connection lifetime (= PSH is not new, we have seen it before)"; or does it mean "new within `monitor-aggregation-interval`" (= PSH is new, it's been a while since we saw it)?
3. The same questions but for `FIN` (asked separately, since after the FIN handshake completes the connection will cease to exist and so what happens to any notifications that might be on a timer?)
4. What notifications, if any, are generated at `medium` level for RST packets which don't (necessarily) belong to any established connection?

Reply:
Great questions, thanks! I think we can iterate further on this patch to clarify these points further.

(1) The first time that the `PSH` flag is seen for the direction, it will generate an event. After the first time that any given flag is seen in a particular direction, the presence of that flag will no longer trigger subsequent events; it will fall back to the time-based trigger.
(2) Yes. "new flags" means "new over the connection lifetime". (This is also checked separately for each direction, which we can highlight in the docs as well).
(3) We've not previously defined this very clearly, but I suspect that it will make sense for FIN to reset the flags & monitor interval logic, so that new connections on the same tuple can be properly monitored. This is not yet the case today in current releases.
(4) If there is no existing connection being tracked, then a monitor notification is always emitted. This includes for all packet drops.